test/e2eTests/extension.e2e.test.ts

```diff
@@ -165,19 +283,31 @@ describe('E2E Test', () => {
         })();
     });
 
-    function assertDiagnostics(expected: { [key: string]: DiagnosticInfo[]; }) {
+    function assertDiagnostics(expected: { [key: string]: DiagnosticInfo[]; }, isRealTimeValidation = false) {
         Object.entries(expected).forEach(([file, expectedDiagnostics]) => {
             const fileUri = Uri.file(path.join(vscode.workspace.workspaceFolders[0].uri.fsPath, "vscode-docs-build-e2e-test", file));
             const diagnostics = vscode.languages.getDiagnostics(fileUri);
-
+            const actualDiagnosticsForCurrentFile: Diagnostic[] = [];

```

Comment:
how about extract a function to the diagnostics with return value type as `Diagnostic[]`, and in this function you can do somethig like:
- get the diagnostics of a file.
- traverse the list and make sure each item is `Diagnostic`, if it not, convert it to `Diagnostic` if it support.
so we can get rid of the parameter `isRealTimeValidation `;

Reply:
I extracted the function in the latest pr. However, diagnostics from real-time validation is ProtocalDiagnostic which is a sub class of Diagnostic. So I always traverse the diagnostics no matter they are from real-time validation or full-repo validation.